ast/statements.go

```diff
@@ -519,3 +551,81 @@ func (e *SetAttr) String() string {
 	out.WriteString(e.value.String())
 	return out.String()
 }
+
+// A Go statement node represents a go statement.
+type Go struct {
+	token token.Token
+	call  Expression
+}
+
```

Comment:
I think it'd be very useful if the `go` keyword was an expression instead of a statement.

And have it return the Risor `thread` object.

Because then you could have:

```go
x := go thing()

x.wait()
```

In other words, it would be semantically identical to:

```go
x := spawn(thing)

x.wait()
```

but with nicer syntax. Because then using Risor to do some quick parallel things becomes super easy. Such as:

```go
a := go fetch("https://url-a")
b := go fetch("https://url-b")
c := go fetch("https://url-c")

my_handling_func(a.wait(), b.wait(), c.wait())
```

And at that point, you don't really need to expose the `spawn` builtin function, but instead only rely on the `go` keyword.

Reply:
I definitely considered this. I do think there's something to be said for matching Go's behavior for the `go` keyword exactly though. It seems like a plus if most or all `go` statements in Risor are also valid in Go. There's an element of making it the least surprising to a Go programmer. This is why I chose to keep it as a statement for now.

Reply:
I'd argue the other way around. That Go syntax and semantics could be expected to work in Risor, but then Risor could add stuff on top to make scripting easier.

Changing from a statement to an expression only expands its usage. So Go semantics are still valid.

Reply:
`risor` is not go and if it provides some functionality that is similar but not exactly the same it only makes things more confusing, I'd even argue it would be better to stick to `spawn` and `thread` and not use the `go` keyword at all.

Reply:
I just find the `spawn` syntax so inconvenient. Similar to Risor's `try`. Switching between function calling syntax is confusing and I commonly trip on it while writing Risor code.

You're fully right @luisdavim that reusing the `go` keyword does bring over a lot of assumptions from the programmer.

Maybe a different keyword could be used, like just using the word `spawn`

```go
spawn(my_func, 1, 2)
spawn my_func(1, 2)
```

Or what if `spawn` was a builtin property on all functions?

```go
my_func(1, 2) // run immediately
my_func.spawn(1, 2) // spawn a goroutine
```

Reply:
I added support for this:

```go
my_func.spawn(1, 2) // spawn a goroutine
```

I did keep the `go` keyword as a statement, so it doesn't return the thread handle. If you want the thread handle you should use `spawn`.

This decision keeps the `go` behavior very similar to Go's for now, and doesn't preclude us from switching it to an expression down the road.